src/debug.rs

```diff
@@ -140,7 +140,8 @@ impl Config {
             .with(
                 tracing_subscriber::fmt::layer()
                     .json()
-                    .with_span_events(FmtSpan::FULL)
+                    .flatten_event(true)
+                    .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
```

Comment:
IMO, these are the span events we really care about. From [the docs](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/struct.Layer.html#method.with_span_events):
> `FmtSpan::NEW`: An event will be synthesized when spans are created.
> `FmtSpan::ENTER`: An event will be synthesized when spans are entered.
> `FmtSpan::EXIT`: An event will be synthesized when spans are exited.
> `FmtSpan::CLOSE`: An event will be synthesized when a span closes.

From my analysis, I noticed that `ENTER` is emitted way more often than it is actually called. For example, I know that the `main` function  in `broker::cmd::run` is only called once. During one sample execution, a `NEW` event was understandably emitted only once, but `ENTER` was emitted 426 times. I suspect this is because a thread is constantly exiting and re-entering spans as it context-switches between different processes.

Reply:
> I suspect this is because a thread is constantly exiting and re-entering spans as it context-switches between different processes.

Correct!

> IMO, these are the span events we really care about.

Good insight, I agree completely. Really, the "new" and "close" events are all we need for latency measures.